source/exec/docker.d

```diff
@@ -44,31 +52,37 @@ class Docker: IExecProvider
 		logInfo("Maximum Queue Size: %d", maximumQueueSize_);
 		logInfo("Memory Limit: %d MB", memoryLimitMB_);
 		logInfo("Output size limit: %d B", maximumQueueSize_);
-		logInfo("Checking whether Docker is functional and updating Docker image '%s'", DockerImage);
-		logInfo("Using docker binary at '%s'", dockerBinaryPath);
 
-		auto docker = execute([this.dockerBinaryPath_, "ps"]);
-		if (docker.status != 0) {
-			throw new Exception("Docker doesn't seem to be functional. Error: '"
-					~ docker.output ~ "'. RC: " ~ to!string(docker.status));
-		}
+		foreach (dockerImage; DockerImages)
+		{
 
-		auto dockerPull = execute([this.dockerBinaryPath_, "pull", DockerImage]);
-		if (docker.status != 0) {
-			throw new Exception("Failed pulling RDMD Docker image. Error: '" ~ docker.output
-					~ "'. RC: " ~ to!string(docker.status));
-		}
+			logInfo("Checking whether Docker is functional and updating Docker image '%s'", dockerImage);
+			logInfo("Using docker binary at '%s'", dockerBinaryPath);
+
+			auto docker = execute([this.dockerBinaryPath_, "ps"]);
+			if (docker.status != 0) {
+				throw new Exception("Docker doesn't seem to be functional. Error: '"
+						~ docker.output ~ "'. RC: " ~ to!string(docker.status));
+			}
 
-		logInfo("Pulled Docker image '%s'.", DockerImage);
-		logInfo("Verifying functionality with 'Hello World' program...");
-		auto result = compileAndExecute(q{void main() { import std.stdio; write("Hello World"); }});
-		enforce(result.success && result.output == "Hello World",
-				new Exception("Compiling 'Hello World' wasn't successful: " ~ result.output));
+			auto dockerPull = execute([this.dockerBinaryPath_, "pull", dockerImage]);
+			if (docker.status != 0) {
```

Comment:
I didn't touch this code - I merely indented it :scream: (so I would leave this for a future improvement).